src/main/java/org/mozilla/zest/core/v1/ZestAssignFromElement.java

```diff
@@ -0,0 +1,251 @@
+        if(element == null){
+            return null;
+        }
+        return getReturnValue(element);
+    }
+
+    private String getReturnValue(Element element) throws ZestAssignFailException {
+        if(returnElement){
+            return element.getContent().toString();
+        }
+        else if(returnAttribute){
+            return element.getAttributeValue(returnedAttributeName);
+        }
+        else {
+            throw new ZestAssignFailException(this, "A selection method must be configured");
+        }
+    }
+
+    private Element findElementAtIndex(List<Element> elements) {
+        if(indexIsOutOfRange(elements)){
+            return null;
+        }
+        return elements.get(elementIndex);
+    }
+
+    private boolean indexIsOutOfRange(List<Element> elements) {
+        return elements.size() <= 0 || elementIndex >= elements.size();
+    }
+
+    private List<Element> reverseIfConfigured(List<Element> elements) {
+        if(reverseFilteredElements){
+            Collections.reverse(elements);
+            return elements;
+        }
+        return elements;
+    }
+
+    private List<Element> filterByElementNameIfConfigured(Source source){
+        if(filterElementName){
+            return source.getAllElements(elementNameFilter);
+        }
+        return source.getAllElements();
+    }
+
+    private List<Element> filterByAttributeValueIfConfigured(List<Element> elements) {
+        if (filterAttributeValue) {
+            List<Element> matchingElements = new ArrayList<>();
+            for (Element element: elements) {
+                if(isMatchingAttributeFilter(element)){
+                    matchingElements.add(element);
+                }
+            }
+            return matchingElements;
+        }
+        return elements;
+    }
+
+    private boolean isMatchingAttributeFilter(Element element) {
+        if(attributeValueFilterPattern == null){
+            attributeValueFilterPattern = Pattern.compile(attributeValueFilter, attributeValueFilterFlags);
+        }
+
+        String attributeValue = element.getAttributeValue(attributeNameFilter);
+        if(attributeValue == null){
+            return false;
+        }
+        Matcher matcher = attributeValueFilterPattern.matcher(attributeValue);
+        return matcher.matches();
+    }
+
+    public ZestAssignFromElement whereElementIs(String elementName){
+        filterElementName = true;
+        elementNameFilter = elementName;
+        return this;
+    }
+
+    public ZestAssignFromElement whereAttributeIs(String attributeName, String attributeValueRegExp) {
+        return whereAttributeIs(attributeName, attributeValueRegExp, 0);
+    }
+
+    public ZestAssignFromElement whereAttributeIs(String attributeName, String attributeValueRegExp, int flags){
+        filterAttributeValue = true;
+        attributeNameFilter = attributeName;
+        attributeValueFilter = attributeValueRegExp;
```

Comment:
Worth resetting the `attributeValueFilterPattern` (just to make sure it's recreated the correct `Pattern` later on).